src/carts/src/main/java/mushop/carts/controllers/CartsController.java

```diff
@@ -0,0 +1,163 @@
+        }
+        return cart;
+    }
+
+    @Get("/{cartId}/items")
+    public List<Item> getCartItems(String cartId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+        return cart.getItems();
+    }
+
+    @Delete("/{cartId}")
+    public Cart deleteCart(String cartId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+
+        if (cartRepository.deleteCart(cartId)) {
+            meterRegistry.counter("carts.deleted").increment();
+            LOG.info("Cart deleted: {}", cart);
+            return cart;
+        } else {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Failed to delete cart " + cartId);
+        }
+    }
+
+    @Delete("/{cartId}/items/{itemId}")
+    public Cart deleteCartItem(String cartId, String itemId) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
+
+        if (!cart.removeItem(itemId)) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart item with id " + itemId + " not found " + cart);
+        }
+
+        meterRegistry.timer("carts.updated.timer").record(() -> {
+            cartRepository.save(cart);
+        });
+        meterRegistry.counter("carts.updated.counter").increment();
+        LOG.info("Item deleted: {}", cart);
+        return cart;
+    }
+
+    @Post("/{cartId}")
+    public HttpResponse<Cart> postCart(@PathVariable String cartId, @Body Cart newCart) {
+
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            newCart.setId(cartId);
+            meterRegistry.timer("carts.created.timer").record(() -> {
+                cartRepository.save(newCart);
+            });
+            meterRegistry.counter("carts.created.counter").increment();
+            LOG.info("Cart created: {}", newCart);
+            return HttpResponse.created(newCart);
+        } else {
+            cart.merge(newCart);
+            meterRegistry.timer("carts.updated.timer").record(() -> {
+                cartRepository.save(cart);
+            });
+            meterRegistry.counter("carts.updated.counter").increment();
+            LOG.info("Cart updated: {}", cart);
+            return HttpResponse.ok(cart);
+        }
+    }
+
+    @Put("/{cartId}/items")
+    public HttpResponse<Cart> updateCartItem(@PathVariable String cartId, @Body Item qItem) {
+        Cart cart = cartRepository.getById(cartId);
+        if (cart == null) {
+            throw new HttpStatusException(HttpStatus.NOT_FOUND, "Cart with id " + cartId + " not found");
+        }
```

Comment:
Simply returning `null` will result in a 404. You can change the method signature to return `Cart` and simplify this quite a b it